sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -226,7 +224,7 @@ private[sql] case class ParquetRelation2(
     private var commonMetadataStatuses: Array[FileStatus] = _
 
     // Parquet footer cache.
-    var footers: Map[FileStatus, Footer] = _
+    var footers: Map[Path, Footer] = _
```

Comment:
`FileStatus` objects are also cached, so this should be OK. Bounding the size can be a good idea.